app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/HedvigBottomSheet.kt

```diff
@@ -25,38 +56,111 @@ import eu.wewox.modalsheet.ModalSheet
 fun HedvigBottomSheet(
   isVisible: Boolean,
   onVisibleChange: (Boolean) -> Unit,
+  bottomButtonText: String,
   onSystemBack: (() -> Unit)? = { onVisibleChange(false) },
+  onBottomButtonClick: () -> Unit = { onVisibleChange(false) },
   sheetPadding: PaddingValues? = null,
   cancelable: Boolean = true,
   content: @Composable ColumnScope.() -> Unit,
 ) {
+  val scrollState = rememberScrollState()
+  var scrollDown by remember { mutableStateOf(false) }
+  LaunchedEffect(scrollDown) {
+    if (scrollDown) {
+      scrollState.animateScrollTo(scrollState.maxValue)
+    }
+  }
   val isImeVisible = WindowInsets.isImeVisible
   val defaultPadding = if (isImeVisible) {
     WindowInsets.ime.asPaddingValues()
   } else {
     WindowInsets.safeDrawing
-      .only(WindowInsetsSides.Bottom).asPaddingValues()
+      .only(WindowInsetsSides.Bottom + WindowInsetsSides.Top).asPaddingValues()
```

Comment:
If you are scrolled down, and you focus on the "Size" entry (in extra buildings screen) you will notice that now this makes the content render behind the top app bar temporarily, and it does not do that when you then dismiss the keyboard.

What are we trying to do here with this check here? Could we just do without any if/else check here?
The `safeDrawing` btw does include the ime padding itself too internally, see here https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/foundation/foundation-layout/src/androidMain/kotlin/androidx/compose/foundation/layout/WindowInsets.android.kt;l=400?q=safeDrawing
The doc reads:
"""
The insets that include areas where content may be covered by other drawn content. This includes
all [system bars][systemBars], [display cutout][displayCutout], and [soft keyboard][ime].
"""

Reply:
I think the imePadding is one of the things that were not working properly previously, I'll check this one out!